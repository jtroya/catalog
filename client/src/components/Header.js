import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Header({ appName }) {
  return (
    <nav>
      <div className="nav-wrapper">
        <Link to="/"><span className="brand-logo center">{ appName }</span></Link>
      </div>
    </nav>
  );
};

Header.propTypes = {
  appName: PropTypes.string  
};

Header.defaultProps = {
  appName: 'Catalog'  
};

export default Header;
