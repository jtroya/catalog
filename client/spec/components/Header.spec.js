import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../../src/components/Header';

describe('Header', () => {
  let wrapper;
  
  it('should render the Header', () => {
    wrapper = shallow(<Header appName="My App" />);
    expect(wrapper.find('Navbar')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = shallow(<Header appName="My App" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
