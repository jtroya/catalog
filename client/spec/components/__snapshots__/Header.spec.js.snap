// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<nav>
  <div
    className="nav-wrapper"
  >
    <Link
      replace={false}
      to="/"
    >
      <span
        className="brand-logo center"
      >
        My App
      </span>
    </Link>
  </div>
</nav>
`;
